Use gDelta.1 for sampling weights

kecolson opened the pull request:
Adjust handling of gDelta.1, so that the user can use this term to specify and apply sampling or matching weights, as was done for tmle for ATE here:
Rudolf KE, Diaz I, Rosenblum M, Stuart EA (2014) Estimating population treatment effects from a survey subsample. Am J Epidemiology, doi: 10.1093/aje/kwu197. 
http://aje.oxfordjournals.org/content/early/2014/09/04/aje.kwu197

kecolson committed:
Use gDelta.1 for sampling weights 

Adjust handling of gDelta.1, so that the user can use this term to specify and apply sampling or matching weights, as was done for tmle for ATE here:
Rudolf KE, Diaz I, Rosenblum M, Stuart EA (2014) Estimating population treatment effects from a survey subsample. Am J Epidemiology, doi: 10.1093/aje/kwu197. 
http://aje.oxfordjournals.org/content/early/2014/09/04/aje.kwu197

lendle commented on R/tmlecte.R:
```diff
@@ -181,7 +181,12 @@ tmle.cte <- function(A, B, Y, a=0, Delta=NULL, Q.method="glm", Q.formula=NULL, Q
     gDelta.1 <- .bound(gDelta.1, c(1, min(gbound)))
   } else {
-   gDelta.1 <- rep(1, length(Y))
+    if (method=="user" && !is.null(gDelta.1) && sum(!is.na(gDelta.1))>0) {
```
Here `method` should be `gDelta.method` I think

kecolson committed:
Correct typo

kecolson commented:
Good catch- your'e right. I think I've corrected it now.

lendle commented:
Can you also add a note about this to the documentation in the roxygen comment? The [`gDelta.1`](https://github.com/kecolson/tmlecte/blob/master/R/tmlecte.R#L70) argument is probably the best place, or maybe the details section, which doesn't currently exist.

kecolson committed:
Add gDelta.1 documentation to roxygen

In roxygen documentation for package, added additional information on use of user-specified gDelta.1 for applying matching or sampling weights.
